tools/mavtomfile.py

```diff
@@ -74,6 +74,8 @@ def process_tlog(filename):
         f.write("%s.data(%u,:) = [%f" % (mtype, type_counters[mtype], m._timestamp))
         for field in m._fieldnames:
             val = getattr(m, field)
+            if isinstance(val,unicode):
```

Comment:
```suggestion
            if isinstance(val, unicode):
```

Reply:
To be honest, I'm not really sure right now. The issue seems to be from how certain .BIN log files are created. I'll be looking into this to offer more information when I can.

Reply:
Hi, my apologies for the delay, but I have been exploring this. I ran a simulation and generated a .BIN file. I then used a debugger to take note which values were being stored as unicode. 

The snippet below is from the txt output of the same .BIN file. This output was generated using MAVlopdump.py.

2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 45, Label : }
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 63, Label : UNKNOWN}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 65, Label : A}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 100, Label : deg}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 98, Label : B}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 107, Label : deg/s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 68, Label : deglatitude}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 101, Label : deg/s/s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 69, Label : rad/s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 71, Label : Gauss}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 104, Label : degheading}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 105, Label : A.s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 74, Label : W.s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 76, Label : rad/s/s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 109, Label : m}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 110, Label : m/s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 111, Label : m/s/s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 79, Label : degC}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 37, Label : %}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 83, Label : satellites}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 115, Label : s}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 113, Label : rpm}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 114, Label : rad}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 85, Label : deglongitude}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 117, Label : ppm}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 118, Label : V}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 80, Label : Pa}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 119, Label : Ohm}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 89, Label : us}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 122, Label : Hz}
2020-08-04 14:37:54.50: UNIT {TimeUS : 1864913736, Id : 35, Label : instance}

All of the values for "Label" are stored as unicode. There are others but those led me to LogStructure.h. There were other strings labeled as unicode that were also found in LogStructure.h (e.g. "s---SmDUmnhn-"). So it's clear that these strings are coming from LogStructure.h however, WHY they're being stored as unicode in the .BIN files I have not been able to figure out. I will continue to explore, but this is what I've found so far.